src/uct/cuda/base/cuda_iface.c

```diff
@@ -11,6 +11,13 @@
 #include "cuda_iface.h"
 
 
+void uct_cuda_base_cu_get_error_string(CUresult result, const char **error_str)
+{
+    if (cuGetErrorString(result, error_str) != CUDA_SUCCESS) {
+        *error_str = "unrecognized error code";
```

Comment:
shall we include the error code in the string?
could use a global/TLS string buffer

Reply:
fixed

Reply:
why need TLS if UCT is not thread safe?

Reply:
we should support using 2 uct workers from different threads, for example